Changes to publish entire interaction session data to ipfs and ipfs-c…

pankaj-nayak opened the pull request:
This PR resolves #11 

This change includes additional interaction session data between AiModel and Interrogator and its corresponding quantum score being published with overall result to public ipfs network and ipfs-cluster.

Unit testing Done : 

VISIT: https://explore.ipld.io/#/explore/bafyreidcgddav3js65stodcqzluxmvypykxgtyluy5eaj4bdkime6sqs5y
for interrogation results. This link is public and the result data is written to the public IPFS network!

VISIT: https://kovan.etherscan.io/tx/0xb989814ab79255b4b59f8bef00f1fd5fef9582360971627b1a4759a1b3319cc5
 for transaction details. This link is public and written to the Ethereum Kovan tesnet!
——————————————————————————————————————————————————

VISIT: https://explore.ipld.io/#/explore/bafyreih4gwhj3yitfwqhwbctcdcv2y23u7rwdv4a3khyiseumubnbm7qk4
for interrogation results. This link is public and the result data is written to the public IPFS network!

VISIT: https://kovan.etherscan.io/tx/0x476beccf8e8cd0a53e62ee64750f8dc11172096509d350fb05e8fd777c3239fa
for transaction details. This link is public and written to the Ethereum Kovan tesnet!

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/texaproject/texa/12)
<!-- Reviewable:end -->

pankaj-nayak committed:
Changes to publish entire interaction session data to ipfs and ipfs-cluster.

This change includes additional interaction session data between AiModel and Interrogator and its correspoinding quantum score being published with overall result to public ipfs network and ipfs-cluster.

0zAND1z commented on go.mod:
```diff
@@ -7,4 +7,6 @@ require (
 	github.com/TexaProject/texajson v0.0.4
 	github.com/TexaProject/texalib v0.0.0-20170504075604-c488f5b05455
 	github.com/ethereum/go-ethereum v1.9.15
+	github.com/ipfs/go-cid v0.0.7
```
Did you run `go mod clean` before making the push?

pankaj-nayak replied:
no. Done now.

0zAND1z commented on main.go:
```diff
@@ -341,6 +364,13 @@ func getSlabJSON(w http.ResponseWriter, r *http.Request) {
 func main() {
+
+	host := flag.String("host", "localhost", "identifier for ipfs and ipfs cluster")
+	port := flag.String("port", "9094", "Source port number")
```
Shouldn't the description be "Port number of ipfs-cluster" instead?

pankaj-nayak replied:
corrected.Done.

0zAND1z commented on go.mod:
```diff
@@ -7,4 +7,6 @@ require (
 	github.com/TexaProject/texajson v0.0.4
```
After the texajson PR, I have rolled out a new release. So please make sure that your `go.mod` points to `v0.0.5`. :)

pankaj-nayak replied:
sure :)

0zAND1z commented on www/index.html:
```diff
@@ -17,6 +17,10 @@
     SlabNum = "";
     while (SlabNum == "" || SlabNum == null || SlabNum == 0) {
       SlabNum = prompt("Enter number of Slabs:", "");
+      if (isNaN(SlabNum) == true) {
```
Can we do similar form/input validations for all inputs?

Only if it's not too much to ask (optional).

pankaj-nayak replied:
we have following input types : 
1. AI NAME : characters and integers are accepted. 
2. IntName : characters and integers are accepted. 
3. Slab Count : Only Numbers are accepted [Check is there.]
4. Slab Name : characters and integers are accepted. 
5. Justification : characters and integers are accepted. 
6. Transactions : characters and integers are accepted.
7. Quantum Score : Only 0 and 1 are accepted [Check is there]

I did't put check for remaining types thinking - input could be of any kind, for example : Slab name could  be 001, 002 for a particular AI model . For justification and transaction user is free to use both numbers and strings. 

Kindly let me know any case which i might be missing ,i will take care of it.

0zAND1z replied:
Yes @pankaj-nayak, it could be an alphanumeric string.

pankaj-nayak committed:
Taking care of review comments

Taking care of review comments.

Unit testing :

VISIT: https://explore.ipld.io/#/explore/bafyreibr342qr7vbwyl3e25x4auqov3yvxldnu7zsiktib2uwr53zcvxie
 for interrogation results. This link is public and the result data is written to the public IPFS network!

VISIT: https://kovan.etherscan.io/tx/0x2a2df7c3eb47beca70a60c30e382e24a4fba3803a57bd8ba1fe3c295b5e50a3e
for transaction details. This link is public and written to the Ethereum Kovan tesnet!
